### DISCUSSION TITLE
doris stream load csv[INTERNAL_ERROR]too many filtered rows]导入报错

### DISCUSSION THREAD

**Discussion by wandayou** [apache/doris#22682] [A - General / Q&A]
导入报错信息：
[root@pc01 ~]# curl --location-trusted -uroot: -T /tmp/datafiles/test_01.csv -H “format:csv_with_names” -H “trim_double_quotes:true” -H “column_separator:,” http://172.16.152.206:8030/api/test/ods_test01/_stream_load
{
“TxnId”: 6203,
“Label”: “a5d05904-26b6-423d-b05f-a140891be62b”,
“Comment”: “”,
“TwoPhaseCommit”: “false”,
“Status”: “Fail”,
“Message”: “[INTERNAL_ERROR]too many filtered rows”,
“NumberTotalRows”: 3,
“NumberLoadedRows”: 2,
“NumberFilteredRows”: 1,
“NumberUnselectedRows”: 0,
“LoadBytes”: 86,
“LoadTimeMs”: 18,
“BeginTxnTimeMs”: 1,
“StreamLoadPutTimeMs”: 5,
“ReadDataTimeMs”: 0,
“WriteDataTimeMs”: 7,
“CommitAndPublishTimeMs”: 0,
“ErrorURL”: “http://172.16.152.207:8040/api/_load_error_log?file=__shard_8/error_log_insert_stmt_574a7a02c4eaa1b0-e2ada45969919599_574a7a02c4eaa1b0_e2ada45969919599”
}

ErrorURL内容：
Reason: actual column number in csv file is more than schema column number.actual number: 4, column separator: [,], line delimiter: [
], schema column number: 2; . src line [1000000601,“科技,大数据,Doris”
];

1.建表语句：
CREATE TABLE IF NOT EXISTS ods_test01
(
ProductID VARCHAR(100),
Keyword VARCHAR(256)
)
Unique KEY(ProductID)
DISTRIBUTED BY HASH(ProductID) BUCKETS 3
PROPERTIES (
“replication_allocation” = “tag.location.default: 3”
);
2.导入test_01.csv数据：
ProductID,Keyword
1000000601,“科技,大数据,Doris”
1000000602,科技
1000000603,

**Comment by wandayou** [comment]
,“科技,大数据,Doris”

csv文件 部分列内容中含有逗号的话，需要用双引号括起来，而实际生产上也是避免不了这种情况。有什么好的解决方法吗？另外在测试flink使用doris-connector向doris表中写入数据时，如果列中含有逗号也会报类似错误

**Comment by zy-kkk** [comment]
可以使用json格式导入，或者更换csv格式导入的行列分隔符，可参考https://doris.apache.org/zh-CN/docs/dev/ecosystem/flink-doris-connector#sink-%E9%85%8D%E7%BD%AE%E9%A1%B9

**Reply by cl0924** [reply]
请问flink-doris-connector是否支持parquet格式数据写入doris，我参照json写了配置'sink.properties.format' = 'parquet' 但尝试格式检验失败了

**Reply by wandayou** [reply]
使用json格式配置确实可以成功导入，eg:
CREATE TABLE ods_contentlist_sink (
	id string,
	type string
) 
WITH (
      'connector' = 'doris',
      'fenodes' = '172.16.152.206:8030',
      'table.identifier' = 'test.ods_contentlist',
      'username' = 'root',
      'password' = '',
      'sink.properties.format' = 'json',
      'sink.properties.read_json_by_line' = 'true',
      'sink.label-prefix' = 'doris_label',
     'sink.enable-2pc'='true'
);

**Comment by wandayou** [comment]
这是一封自动回复邮件。已经收到您的来信，我会尽快回复。谢谢！
                                                              ---万大友
